src/main/java/uk/gov/nationalarchives/pdi/step/xml/CanonicalStepDialog.java

```diff
@@ -0,0 +1,283 @@
+        // the dialog, it will be restored to this saved value.
+        // The "changed" variable is inherited from BaseStepDialog
+        changed = meta.hasChanged();
+
+        // The ModifyListener used on all controls. It will update the meta object to
+        // indicate that changes are being made.
+        ModifyListener lsMod = new ModifyListener() {
+            public void modifyText( ModifyEvent e ) {
+                meta.setChanged();
+            }
+        };
+
+        // ------------------------------------------------------- //
+        // SWT code for building the actual settings dialog        //
+        // ------------------------------------------------------- //
+        FormLayout formLayout = new FormLayout();
+        formLayout.marginWidth = Const.FORM_MARGIN;
+        formLayout.marginHeight = Const.FORM_MARGIN;
+        shell.setLayout( formLayout );
+        shell.setText( BaseMessages.getString( PKG, "CanonicalStep.Shell.Title" ) );
+        int middle = props.getMiddlePct();
+        int margin = Const.MARGIN;
+
+        // Stepname label
+        wlStepname = new Label( shell, SWT.RIGHT );
+        wlStepname.setText( BaseMessages.getString( PKG, "System.Label.StepName" ) );
+        props.setLook( wlStepname );
+        fdlStepname = new FormData();
+        fdlStepname.left = new FormAttachment( 0, 0 );
+        fdlStepname.right = new FormAttachment( middle, -margin );
+        fdlStepname.top = new FormAttachment( 0, margin );
+        wlStepname.setLayoutData( fdlStepname );
+
+        // Stepname input box
+        wStepname = new Text( shell, SWT.SINGLE | SWT.LEFT | SWT.BORDER );
+        wStepname.setText( stepname );
+        props.setLook( wStepname );
+        wStepname.addModifyListener( lsMod );
+        fdStepname = new FormData();
+        fdStepname.left = new FormAttachment( middle, 0 );
+        fdStepname.top = new FormAttachment( 0, margin );
+        fdStepname.right = new FormAttachment( 100, 0 );
+        wStepname.setLayoutData( fdStepname );
+
+        // FieldName to evaluate
+        Label wlField = new Label(shell, SWT.RIGHT);
+        wlField.setText( BaseMessages.getString( PKG, "CanonicalStep.Input.Field.Label" ) );
+        props.setLook(wlField);
+        FormData fdlField = new FormData();
+        fdlField.left = new FormAttachment( 0, 0 );
+        fdlField.top = new FormAttachment( wStepname, 2 * margin );
+        fdlField.right = new FormAttachment( middle, -margin );
+        wlField.setLayoutData( fdlField );
+        xmlInputField = new CCombo( shell, SWT.BORDER | SWT.READ_ONLY );
+        xmlInputField.setEditable( true );
+        props.setLook(xmlInputField);
+        xmlInputField.addModifyListener( lsMod );
+
+        FormData fdField = new FormData();
+        fdField.left = new FormAttachment( middle, margin );
+        fdField.top = new FormAttachment( wStepname, 2 * margin );
+        fdField.right = new FormAttachment( 100, -margin );
```

Comment:
Can we define some constants for things like `100`. e.g. See - `LABEL_SPACING` and `WIDTH` in https://github.com/nationalarchives/kettle-jena-plugins/blob/main/src/main/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepDialog.java#L56